src/citrine/informatics/predictors/auto_ml_predictor.py

```diff
@@ -51,28 +54,52 @@ def __init__(self,
                  name: str,
                  *,
                  description: str,
-                 output: Descriptor,
+                 output: Descriptor = None,
+                 outputs: List[Descriptor] = None,
                  inputs: List[Descriptor],
                  training_data: Optional[List[DataSource]] = None):
         self.name: str = name
         self.description: str = description
         self.inputs: List[Descriptor] = inputs
-        self.output: Descriptor = output
         self.training_data: List[DataSource] = training_data or []
 
+        if output is not None:
+            msg = ('The "output" parameter is deprecated as of 1.24.0 and will be removed in '
+                   '2.0.0. Please use the "outputs" field instead.')
+            warnings.warn(msg, category=DeprecationWarning)
+            if outputs is not None:
+                raise ValueError('Found values for "outputs" and the deprecated "output". Please '
+                                 'provide only "outputs".')
+
+        self.outputs: List[Descriptor] = []
+        if outputs is not None:
+            self.outputs = outputs
+        elif output is not None:
+            self.outputs = [output]
+
     def _post_dump(self, data: dict) -> dict:
         data['display_name'] = data['config']['name']
-        data['config']['outputs'] = [data['output']]
-        data['config']['responses'] = [data['output']]
         return data
 
-    @classmethod
-    def _pre_build(cls, data: dict) -> dict:
-        if 'outputs' in data['config']:
-            data['output'] = data['config']['outputs'][0]
-        elif 'responses' in data['config']:
-            data['output'] = data['config']['responses'][0]
-        return data
+    @property
+    def output(self) -> Optional[Descriptor]:
+        """[DEPRECATED] Get the first output descriptor."""
+        msg = ('The "output" field is deprecated as of 1.24.0 and will be removed in 2.0.0. '
+               'Please use the "outputs" field instead.')
+        warnings.warn(msg, category=DeprecationWarning)
+        return self.outputs[0] if self.outputs else None
+
+    @output.setter
+    def output(self, value):
+        """[DEPRECATED] Set the output descriptor."""
+        msg = ('The "output" field is deprecated as of 1.24.0 and will be removed in 2.0.0. '
+               'Please use the "outputs" field instead.')
+        warnings.warn(msg, category=DeprecationWarning)
+        if not isinstance(value, Descriptor):
+            # Mirroring the error emited by serialization.
+            raise ValueError(f"{value} is not one of valid types: <class 'dict'> for output")
```

Comment:
Shouldn't this be `Descriptor` instead of `dict`? Both because the serialization error prints the underlying type, and also because it makes more sense given what the user is trying to do here.

Reply:
That was also my assumption. However, this is the error message I get from the existing serialization mechanism, so I wanted to keep it constant. Maybe it indicates an erroneous error message in our serialization code?

Reply:
I'd say it's... not a bug, but a misfire in the setter code.

The [setter code](https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/_serialization/properties.py#L145-L156) first checks if the value is of the correct underlying type (in this case, a `Descriptor`). If not, then it assumes the user has passed in the serialized version of the type (which is a `dict`) and it tries to deserialize it. That's why we get an error message saying "I expected a `dict`".

I think that a better flow would be
```
If value is an underlying type:
  set value
else if value is a serialized type:
  deserialize and set value
else:
  throw a ValueError that lists both the underlying and serialized types as possibilities
```

Regardless of whether or not you feel like changing that, we don't need to be held hostage by previous oversights. Might as well make _this_ message more clear.